3 small fixes for runners/orchestration

terminalmage opened the pull request:
This fixes the following issues:

1. The orchestration jid was not being passed to the `state.orchestrate` runner when invoked using the `state.orch` alias. This did not affect the runner's performance/results, but merely kept the jid from being included in the metadata for the job in the job cache.

2. Fix an edge case when orchestration attempts to compile pilllar data before making any other fileclient requests. (Resolves #46004)

3. Keeps salt-run's opts from being modified while running the `pillar.show_pillar` runner.

Additionally, it makes a small performance/memory improvement in the state compiler by extending lists rather than using `+=`, and documents some previously-undocumented behavior when orchestration is invoked using an SLS file containing state SLS instead of orchestration SLS.

terminalmage committed:
Properly detect use of the state.orch alias and add orch jid to kwargs

Before, only when `salt-run state.orchestrate` was used would we add the
orch jid to the kwargs.

terminalmage committed:
salt.runners.pillar.show_pillar: don't modify master opts

This keeps us from modifying the master opts when compiling pillar data
in this runner.

terminalmage committed:
Instantiate the SaltCacheLoader's fileclient in the __init__

This fixes an edge case when orchestration attempts to compile pillar
data before making any other fileclient requests.

terminalmage committed:
Document behavior when orchestration runnner invoked with non-orch states

This also replaces some += usage with list.extend for a modest
performance/memory improvement.

terminalmage commented:
Some legit failures here, I'll take care of them.

terminalmage committed:
Update tests to reflect changes to the SaltCacheLoader

terminalmage commented:
OK, should be fixed now.

github-abcde commented on salt/runners/pillar.py:
```diff
@@ -82,16 +83,18 @@ def show_pillar(minion='*', **kwargs):
     for key in kwargs:
         if key == 'pillarenv':
-            __opts__['pillarenv'] = kwargs[key]
+            opts['pillarenv'] = kwargs[key]
```
About this...this was introduced in https://github.com/ogd-software/salt/commit/8d19790cdab909042b8ded1b5321e8f7e9f4945d but the check for `key == 'pillarenv'` now happens twice...one to modify opts and one to set the `pillarenv` variable which is also passed into `salt.pillar.Pillar`. Why is it neccesary to modify `opts['pillarenv']`  here (in favour of the `pillarenv` variable which is also passed)?

terminalmage replied:
Yeah, you're right... I didn't give this a close enough look. This is fixed in https://github.com/saltstack/salt/pull/46145/commits/2a185855ea89ac5bd9a87a80914cfb5a81d9cf7e

terminalmage replied:
Also added 467ff841 (I had forgotten to save before committing and still had one minor change that should have been in 2a18585).

terminalmage committed:
Better solution for fixing the opts munging in pillar.show_pillar runner

The pillarenv argument to the `Pillar` class is only used as an override
of what is in the opts. Therefore, it should default to None unless
passed in the kwargs.

This also removes the need for a deepcopy of the opts, since we're no
longer making changes to the opts before passing them to the `Pillar`
constructor.

terminalmage committed:
pillarenv argument should default to None and not the value from opts
